pkg/controller/association/controller/beat_kibana.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package controller
+
+import (
+	beatv1beta1 "github.com/elastic/cloud-on-k8s/pkg/apis/beat/v1beta1"
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/association"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/operator"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/watches"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/user"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/kibana"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/rbac"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func AddBeatKibana(mgr manager.Manager, accessReviewer rbac.AccessReviewer, params operator.Parameters) error {
+	return association.AddAssociationController(mgr, accessReviewer, params, association.AssociationInfo{
+		AssociationObjTemplate: func() commonv1.Association { return &beatv1beta1.BeatKibanaAssociation{} },
+		ElasticsearchRef:       getElasticsearchFromKibana,
+		ExternalServiceURL:     getKibanaExternalURL,
+		AssociatedNamer:        kibana.Namer,
+		AssociationName:        "beat-kibana",
+		AssociatedShortName:    "beat",
+		AssociationLabels: func(associated types.NamespacedName) map[string]string {
+			return map[string]string{
+				BeatAssociationLabelName:      associated.Name,
+				BeatAssociationLabelNamespace: associated.Namespace,
+				BeatAssociationLabelType:      commonv1.KibanaAssociationType,
+			}
+		},
+		UserSecretSuffix:  "beat-kb-user",
+		CASecretLabelName: kibana.KibanaNameLabelName,
+		ESUserRole: func(commonv1.Associated) (string, error) {
+			return user.KibanaAdminBuiltinRole, nil
```

Comment:
Will this work with <7.5 version of Kibana? I don't see it in the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/built-in-roles.html).

Reply:
Good point. I forgot about that.